From a71653b20b889a71c6643171603e65b0ca945ea4 Mon Sep 17 00:00:00 2001
From: Frantisek Sumsal <fsumsal@redhat.com>
Date: Wed, 30 Nov 2022 15:24:33 +0100
Subject: [PATCH] test: disable flaky subtests that require udevadm wait/lock

Certain TEST-64-UDEV-STORAGE tests require `udevadm wait` and `udevadm
lock` verbs to work reliably. Since we don't plan to backport the verbs
to RHEL 9.1 and older, let's skip the tests to make CIs a bit more
reliable when dealing with z-streams.

Necessary only on RHEL 9.1 and 9.0.

rhel-only
Related: #2149074
---
 test/TEST-64-UDEV-STORAGE/test.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/test/TEST-64-UDEV-STORAGE/test.sh b/test/TEST-64-UDEV-STORAGE/test.sh
index 2a5b5b2dd3..659b03e07b 100755
--- a/test/TEST-64-UDEV-STORAGE/test.sh
+++ b/test/TEST-64-UDEV-STORAGE/test.sh
@@ -310,6 +310,11 @@ EOF
 
 # Test case for issue https://github.com/systemd/systemd/issues/19946
 testcase_simultaneous_events() {
+    if ! "$BUILD_DIR/udevadm" lock --version >/dev/null; then
+        echo "This test is flaky without udevadm wait/lock, skipping the test..."
+        return 77
+    fi
+
     local qemu_opts=("-device virtio-scsi-pci,id=scsi")
     local partdisk="${TESTDIR:?}/simultaneousevents.img"
 
